app/population/views/edit/edit.njk

```diff
@@ -0,0 +1,32 @@
+{% extends "form-wizard.njk" %}
+
+{% macro fieldGroup(name) %}
+  {%  set key = "population::daily_view.labels." + name %}
+  {%  set label = t(key) %}
+  <h2 class="govuk-heading-m">{{ label}}</h2>
+
+  {% set fieldsForGroup = options.fields | filterByPropertyArray('id', options.fieldGroups[name]) %}
+  {% for key, fieldsOptions in fieldsForGroup %}
+    {% if fieldsOptions.component and not fieldsOptions.skip %}
+      {{ callAsMacro(fieldsOptions.component)(fieldsOptions) }}
+    {% endif %}
+  {% endfor %}
+{% endmacro %}
+
+
+{% block fields %}
+
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full govuk-!-margin-bottom-2">
```

Comment:
I wonder if this could be tackled without the need for that filter:

```nunjucks
{% extends "form-wizard.njk" %}

{% block fields %}
  <div class="govuk-grid-row">
    {% for group, fields in options.fieldGroups %}
      {% set key = "population::daily_view.labels." + group %}

      <div class="govuk-grid-column-full govuk-!-margin-bottom-2">
        <h2 class="govuk-heading-m">{{ t(key) }}</h2>

        {% for fieldKey in fields %}
          {% set field = options.fields[fieldKey] %}

          {% if field.component and not field.skip %}
            {{ callAsMacro(field.component)(field) }}
          {% endif %}
        {% endfor %}
      </div>
    {% endfor %}
  </div>
{% endblock %}
```

Reply:
I like the look of that - much less code to think about!

I think it will still need to be a macro though, as the subsequent PR adds in the Transfers In / Transfers Out - so we'd need to explicitly call each group to be able to add them into the bottom of each `unavailable` / `available` block.

Reply:
Or you just manually add the markup for those elements as it looks like they're custom anyway?

Reply:
So I've dealt with it by using your code as a `macro` - but in #1224 , I'll invoke the macros using `call` - that way I can fill in the body content inside each block, keeping everything contained.